team-cost-reporter/plugins/dynamodb/__init__.py

```diff
@@ -198,4 +225,66 @@ def getTeamCost(team_name,configMap,debug):
                     team_cost['shared'][cc_table_name] = format(float(totalCost),'.2f')
                     totalCost = 0
 
+    elif data and 'aws_ce' in config_plugin:
+        tag_to_match = None
+        if debug: log("%i tags returned" % len(data['Groupings']))
+
+        # Find our team info in the config file
+        for team in configMap['teams']:
+            if team['name'] == team_name:
+                tag_to_match = team[id()]['include_tag']
+                config_envs = team[id()]['tables']
+
+        for config_match_env in config_envs:
+            log("looking for dynamodb owner %s in report data" % config_match_env)
+
+            for cc_env in data['Groupings']:
+                tag_value = cc_env['Name'].split(tag_to_match)
+                if debug: log("tag_value: " + str(tag_value))
+
+                if len(tag_value) == 2:
+                    cc_env_name = str(tag_value[1].strip().lower())
+                else:
+                    continue
+
+                if debug: log("dynamodb table in report data is %s" % cc_env_name)
+
+                # See if we're dealing with a wildcard or exact match
+                match = False
+                if str(config_match_env).startswith("*") and str(config_match_env).endswith("*"):
```

Comment:
All of this wildcard detection code stuff is common, it's just the string you're looking at is different - could you create a function that takes an input string, does this check and then returns True/False?

Reply:
Done